File: LeetCode_Algorithm/455(Assign Cookies)/455.js
var findContentChildren = function(g, s) {
  // sol1 111ms 56% 44.2MB 91%
    g.sort((a,b) => {return a-b})
    s.sort((a,b) => {return a-b})
    let gIdx = 0; let sIdx = 0; let ans = 0
    while (gIdx < g.length && sIdx < s.length) {
      if (g[gIdx] <= s[sIdx]) {
        gIdx += 1; ans += 1
      }
      sIdx += 1
    }
    return ans
};

File: LeetCode_Algorithm/455(Assign Cookies)/455.py
'''
LeetCode  455 : Assign Cookies

# sol1 175ms 67% 15.7MB 79%
O(nlogn) O(n)
정렬 후 크기가 크면 두 인덱스를 같이 증가시키고 아니라면 
쿠키의 인덱스만 증가시킨다.
'''
class Solution:
    def findContentChildren(self, g: list[int], s: list[int]) -> int:
        g.sort(); s.sort()
        g_i = 0; s_i = 0; ans = 0
        while g_i < len(g) and s_i < len(s):
            if g[g_i] <= s[s_i]:
                g_i += 1; s_i += 1; 
                ans += 1;
            else:
                s_i += 1
        return ans
